schemas/layer-4.cue

```diff
@@ -1,37 +1,97 @@
 package schemas
 
+import "time"
+
 #Layer4: {
-    evaluations: [#ControlEvaluation, ...#ControlEvaluation]
+    // ID of the Layer 2 Catalog being evaluated in this evaluation
+    catalog_id: string @go(CatalogID)
+    // timestamp of when the evaluation execution began. If the field is not provided, the evaluation has not been executed yet.
+    start_time?: time.Time @go(StartTime)
+    // timestamp of when the evaluation execution ended. If the field is not provided, the evaluation has not been executed yet.
+    end_time?: time.Time @go(EndTime)
+    // will be true when the evaluation execution changed the evaluated service and could not successfully revert
+    corrupted_state: bool @go(CorruptedState)
+    // one or more evaluations of the framework controls
+    evaluations: [#ControlEvaluation, ...#ControlEvaluation] @go(ControlEvaluations)
 }
 
-// Types
+// URL describes a specific subset of URLs of interest to the framework
+// TODO: this definition should be imported from a more appropriate module/package
+#URL: =~"^https?://[^\\s]+$"
 
+// ControlEvaluation describes the evaluation of the layer 2 control referenced by controlID and the assessment of that control's requirements.
 #ControlEvaluation: {
+    // ID of the layer 2 control being evaluated
+    control_id: string @go(ControlID)
+    // one or more assessments for each of the control's requirements
+    assessments: [#Assessment, ...#Assessment]
+}
+
+// Assessment describes the evaluation of layer 2 control requirement referenced by requirementID and the assessment methods used to assess that requirement.
+#Assessment: {
+    // ID of the Layer 2 Control's Requirement being evaluated
+    requirement_id: string @go(RequirementID)
+    // the methods used to assess the requirement
+    methods: [#AssessmentMethod, ...#AssessmentMethod]
+}
+
+// AssessmentMethod describes the method used to assess the layer 2 control requirement referenced by requirementID.
+#AssessmentMethod: {
+    // Name is the name of the method used to assess the requirement.
     name: string
-    "control-id": string
-    result: #Result
-    message: string
-    "documentation-url"?: =~"^https?://[^\\s]+$"
-    "corrupted-state"?: bool
-    "assessment-results"?: [...#AssessmentResult]
+    // Description is a detailed explanation of the method.
+    description?: string
+    // URL to documentation that describes how the assessment method evaluates the control requirement.
+    documentation?: #URL
+    // Remediation guide is a URL to remediation guidance associated with the control's assessment requirement and this specific assessment method.
+    remediation_guide?: #URL @go(RemediationGuide)
+    // Run is a boolean indicating whether the method was run or not. When run is true, result is expected to be present.
+    run: bool
 }
 
-#AssessmentResult: {
-    result: #Result
+// See https://cuelang.org/docs/tour/types/sumstruct/
+#AssessmentMethod: {
```

Comment:
Perhaps we could an optional `id` field here. I think that could allow an external policy identifier to be used to perform the check in the event a direct check with the method executor is not used. Thoughts?